### ISSUE TITLE
Filtering/authorizing main/parent resource based on relationship's value

### ISSUE THREAD

**Issue by marojeee** [doug-martin/nestjs-query#1329]
I'll use example from the docs: single Task can have multiple SubTasks.

Based on documentation, we can easily filter out SubTasks, when trying to retrieve a Task with its relation to the SubTasks, e.g.:

**TaskDTO:**
```js
@FilterableRelation('subTasks', () => SubTaskDto, {
  auth: {
    authorize: (context: UserContext) => ({
      owner_id: { eq: context.req.user.id },
    }),
  },
})
export class TaskDto {}
```

**SubTaskDTO:**
```js
@FilterableRelation('task', () => TaskDto)
export class SubTaskDto {
  @FilterableField()
  owner_id: number;
}
```

So, with the code above, if there are subtasks with different owners, and all of them belong to the same Task, only those who belong to the owner/logged in user will be returned. And this works great.

---
But, what if we are trying to query SubTasks (as main resource) and then filter it based on Task's ownership, to which is related. Here are DTOs:

**TaskDTO:**
```js
@FilterableRelation('subTasks', () => SubTaskDto)
export class TaskDto {
  owner_id: number;
}
```

**SubTaskDTO:**
```js
@FilterableRelation('task', () => TaskDto)
export class SubTaskDto {}
```

I would like to be able to do something like this:

**SubTaskDTO:**
```js
@Authorize({
  authorize: (context: UserContext) => ({
    task.owner_id: { eq: context.req.user.id },
  }),
})
@FilterableRelation('task', () => TaskDto)
export class SubTaskDto {}
```

In other words, when querying for SubTasks, I would like to get only SubTasks, whose Task's owner is not currently logged in user (`SubTask->Task->owner_id`).

Is something like this already possible? If yes, sorry for (obviously) bad reading of documentation 😅 
I know that it says that complexity can be much bigger than examples provided, but I didn't find a way how to do this, so any help would be appreciated!

P.s. great tool, keep it going 👍

**Comment by doug-martin** [comment]
@marojeee thanks for the question! You should be able to do this doing something like the following 

```
@Authorize({
  authorize: (context: UserContext) => ({
    task: { owner_id: { eq: context.req.user.id } },
  }),
})
```

Let me know if you still have issues after trying the above.
